### PR TITLE
Allow enhanced information for a single named reference

### PR DESCRIPTION
**Pull request by snazy**
This is a planned follow-up on top of #2726 to inquire additional information for a single named reference:
* retrieving the commit-meta of the named reference's HEAD
* retrieving the common-ancestor of the named reference and the default branch
* retrieving the number of commits behind + ahead relative to the common-ancestor

### CODE REVIEW

File: versioned/spi/src/main/java/org/projectnessie/versioned/VersionStore.java [hunk 1]
```diff
@@ -208,6 +194,22 @@ Hash create(NamedRef ref, Optional<Hash> targetHash)
   void delete(NamedRef ref, Optional<Hash> hash)
       throws ReferenceNotFoundException, ReferenceConflictException;
 
+  /**
+   * Provide the current hash for the given NamedRef.
```
Comment (nastra) [inline versioned/spi/src/main/java/org/projectnessie/versioned/VersionStore.java:198]: maybe `Resolve the current hash for the given NamedRef`?

File: versioned/tests/src/main/java/org/projectnessie/versioned/tests/CommitBuilder.java [hunk 2]
```diff
@@ -167,9 +168,12 @@ public Hash toBranch(String branchName)
    */
   public Hash toBranch(BranchName branchName)
       throws ReferenceNotFoundException, ReferenceConflictException {
-    Optional<Hash> reference = fromLatest ? Optional.of(store.toHash(branchName)) : referenceHash;
+    Optional<Hash> reference =
+        fromLatest
+            ? Optional.of(store.getNamedRef(branchName, GetNamedRefsParams.DEFAULT).getHash())
```
Comment (nastra) [inline versioned/tests/src/main/java/org/projectnessie/versioned/tests/CommitBuilder.java:173]: `store.getNamedRef(branchName, GetNamedRefsParams.DEFAULT).getHash()` is being used twice, maybe extract into a variable?

Comment (snazy) [reply versioned/tests/src/main/java/org/projectnessie/versioned/tests/CommitBuilder.java]: updated

File: versioned/spi/src/main/java/org/projectnessie/versioned/VersionStore.java (hunk 1 above)
Comment (snazy) [reply versioned/spi/src/main/java/org/projectnessie/versioned/VersionStore.java]: changed the comment

**Review by nastra — DISMISSED** [review]

**Review by nastra — APPROVED** [review]

**Pull request merged**